common/domain/handler.go

```diff
@@ -594,21 +588,56 @@ func (d *handlerImpl) DeprecateDomain(
 		return err
 	}
 
+	isGlobalDomain := getResponse.IsGlobalDomain
+	if isGlobalDomain && !d.clusterMetadata.IsMasterCluster() {
+		return errNotMasterCluster
+	}
 	getResponse.ConfigVersion = getResponse.ConfigVersion + 1
 	getResponse.Info.Status = persistence.DomainStatusDeprecated
+	lastUpdatedTime := time.Unix(0, getResponse.LastUpdatedTime)
+	now := d.timeSource.Now()
+	// Check the failover cool down time
+	if lastUpdatedTime.Add(d.config.FailoverCoolDown(getResponse.Info.Name)).After(now) {
```

Comment:
not sure if this is useful for domain deprecation... @yux0 What's your opinion on this?

Reply:
Ok, I think it should be removed.